packages/sdk-middleware-http/src/http.js

```diff
@@ -100,17 +100,19 @@ export default function createHttpMiddleware({
   ) => {
     let abortController: any
     const url = host.replace(/\/$/, '') + request.uri
-    const body =
-      typeof request.body === 'string' || Buffer.isBuffer(request.body)
-        ? request.body
-        : // NOTE: `stringify` of `null` gives the String('null')
-          JSON.stringify(request.body || undefined)
-
     const requestHeader: HttpHeaders = { ...request.headers }
     if (!Object.prototype.hasOwnProperty.call(requestHeader, 'Content-Type')) {
       requestHeader['Content-Type'] = 'application/json'
     }
-    if (body) {
+    if(request.headers["Content-Type"] === null) {
+      delete requestHeader['Content-Type']
+    }
+    const body = requestHeader['Content-Type'] === 'application/json'
+        // NOTE: `stringify` of `null` gives the String('null')
+        ? JSON.stringify(request.body || undefined)
+        : request.body
```

Comment:
The failing integration tests might be related to this change. If you want to debug the error, print out the `console.log(body)` to see what value it is.